src/core/resource-container-helper.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {rethrowAsync} from './rethrow-async';
+
+const AMP_CLASS = 'i-amphtml-element';
+const DEEP = true;
+
+const ensureLoaded = (element) => element.ensureLoaded();
+const pause = (element) => element.pause();
+const unmount = (element) => element.unmount();
+
+/**
+ * Ensure all elements within this container are scheduled to load.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function loadAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, !DEEP, ensureLoaded);
+}
+
+/**
+ * Pause all elements within this container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function pauseAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, DEEP, pause);
+}
+
+/**
+ * Unmount all elements within this container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function unmountAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, DEEP, unmount);
+}
+
+/**
+ * Execute a callback for all elements within the container.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean} includeSelf
+ * @param {boolean} deep
+ * @param {function(!AmpElement)} callback
+ */
+export function forAllWithin(
+  containerOrContainers,
+  includeSelf,
+  deep,
+  callback
+) {
+  if (Array.isArray(containerOrContainers)) {
+    for (let i = 0; i < containerOrContainers.length; i++) {
+      forAllWithinInternal(
+        containerOrContainers[i],
+        includeSelf,
+        deep,
+        callback
+      );
+    }
+  } else {
+    forAllWithinInternal(containerOrContainers, includeSelf, deep, callback);
+  }
+}
+
+/**
+ * Execute a callback for all elements within the container.
+ *
+ * @param {!Element} container
+ * @param {boolean} includeSelf
+ * @param {boolean} deep
+ * @param {function(!AmpElement)} callback
+ */
+function forAllWithinInternal(container, includeSelf, deep, callback) {
+  if (includeSelf && container.classList.contains(AMP_CLASS)) {
```

Comment:
We don't actually have `AmpElement` class - it's an interface, so we can't do `instanceof`. However, there are definitely other ways to test a single instance since the `AmpElement` interface has distinctive APIs such as `build()` method. However, another consideration is that this is done for consistency with the descendants search which is done via `getElementsByClassName(sameClass)`, which is equivalent to testing the class name.